app/server/appsmith-server/src/main/java/com/appsmith/server/services/ce/LayoutCollectionServiceCEImpl.java

```diff
@@ -258,14 +262,14 @@ public Mono<LayoutDTO> refactorCollectionName(RefactorActionCollectionNameDTO re
         Mono<String> branchedPageIdMono = StringUtils.isEmpty(branchName)
                 ? Mono.just(pageId)
                 : newPageService
-                .findByBranchNameAndDefaultPageId(branchName, pageId, MANAGE_PAGES)
+                .findByBranchNameAndDefaultPageId(branchName, pageId, pagePermission.getEditPermission())
```

Comment:
Suggestion to change the permission with which we fetch the page while refactoring the collection name. This is to account for scenarios where a user has been given edit access on all the actions of the page but the page itself has a view only access.
```suggestion
                .findByBranchNameAndDefaultPageId(branchName, pageId, pagePermission.getReadPermission())
```